Added WAITING_FOR_AUTH frame

MikeBishop opened the pull request:
Enables multiple requests to point to a single cert challenge.

MikeBishop committed:
Added WAITING_FOR_AUTH frame

Enables multiple requests to point to a single cert challenge.

martinthomson commented on draft-thomson-http2-client-certs.md:
```diff
@@ -185,26 +191,18 @@ client authentication.  In TLS 1.3 a server does this by sending a new TLS 1.3
+populated by the server with a random value.  The server SHOULD then send a
+WAITING_FOR_AUTH frame (see {{frame}}) on any streams where the server is
+awaiting client authentication before responding.  This allows a client to
+correlate the TLS CertificateRequest with one or more outstanding requests.
```
I think that this needs to be a MUST send WAITING_FOR_AUTH and that it has to come before the CertificateRequest, for the reasons that Andrei discussed.

martinthomson commented on draft-thomson-http2-client-certs.md:
```diff
@@ -185,26 +191,18 @@ client authentication.  In TLS 1.3 a server does this by sending a new TLS 1.3
 before sending another TLS CertificateRequest for the same stream.
+CertificateRequest messages for other streams MAY be sent without waiting.  The
+server SHOULD send a new WAITING_FOR_AUTH frame referencing each new
+CertificateRequest.
```
Again, before.  And I think that we need to make it very clear that we need a WAITING_FOR_AUTH for each certificate requested.

martinthomson commented on draft-thomson-http2-client-certs.md:
```diff
@@ -237,7 +235,7 @@ presence of the client certificate presented during the last TLS handshake.
 ## The TLS application_context_id Hello Extension {#extension}
 
-The `application_context_id` TLS Hello Extension is used to carry an idenfier
+The `application_context_id` TLS Hello Extension is used to carry an identifier
```
Thanks for catching all my formatting mistakes and typos :)

MikeBishop committed:
MUST, ordering

Per Martin and Andrei's feedback, upgrading SHOULD to MUST and reversing
the ordering.

martinthomson commented:
I just realized that you updated this.  Merging.  I'll have to tweak a few things later.
